lib/web_ui/test/composition_test.dart

```diff
@@ -0,0 +1,252 @@
+      });
+    });
+  });
+
+  group('composing range', () {
+    late GloballyPositionedTextEditingStrategy editingStrategy;
+
+    setUp(() {
+      editingStrategy = _enableEditingStrategy(deltaModel: false);
+    });
+
+    tearDown(() {
+      editingStrategy.disable();
+    });
+
+    test('should be [0, compostionStrLength] on new composition', () {
+      const String composingText = 'hi';
+
+      _inputElement.dispatchEvent(html.CompositionEvent(_MockWithCompositionAwareMixin._kCompositionUpdate, data: composingText));
+
+      // Set the selection text.
+      _inputElement.value = composingText;
+      _inputElement.dispatchEvent(html.Event.eventType('Event', 'input'));
+
+      expect(
+          editingStrategy.lastEditingState,
+          isA<EditingState>()
+              .having((EditingState editingState) => editingState.composingBaseOffset,
+                  'composingBaseOffset', 0)
+              .having((EditingState editingState) => editingState.composingExtentOffset,
+                  'composingExtentOffset', composingText.length));
+    });
+
+    test(
+        'should be [beforeComposingText - composingText, compostionStrLength] on composition in the middle of text',
+        () {
+      const String composingText = 'hi';
+      const String beforeComposingText = 'beforeComposingText';
+      const String afterComposingText = 'afterComposingText';
+
+      // Type in the text box, then move cursor to the middle.
+      _inputElement.value = '$beforeComposingText$afterComposingText';
+      _inputElement.setSelectionRange(beforeComposingText.length, beforeComposingText.length);
+
+      _inputElement.dispatchEvent(html.CompositionEvent(
+          _MockWithCompositionAwareMixin._kCompositionUpdate,
+          data: composingText));
+
+      // Flush editing state (since we did not compositionend).
+      _inputElement.dispatchEvent(html.Event.eventType('Event', 'input'));
+
+      expect(
+          editingStrategy.lastEditingState,
+          isA<EditingState>()
+              .having((EditingState editingState) => editingState.composingBaseOffset!,
+                  'composingBaseOffset', beforeComposingText.length - composingText.length)
+              .having((EditingState editingState) => editingState.composingExtentOffset,
+                  'composingExtentOffset', beforeComposingText.length));
+    });
+  });
+
+  group('Text Editing Delta Model', () {
+    late GloballyPositionedTextEditingStrategy editingStrategy;
+
+    final StreamController<TextEditingDeltaState?> deltaStream =
+        StreamController<TextEditingDeltaState?>.broadcast();
+
+    setUp(() {
+        editingStrategy = _enableEditingStrategy(
+          deltaModel: true,
+          onChange: (_, TextEditingDeltaState? deltaState) => deltaStream.add(deltaState)
+        );
+    });
+
+    tearDown(() {
+      editingStrategy.disable();
+    });
+
+    test('should have newly entered composing characters', () async {
```

Comment:
@Renzo-Olivares This and the test below it are the tests for your delta fix - lmk your input

Reply:
LGTM thanks for making these